src/selectors/entities/utils.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {getMyChannelMemberships, getAllDirectChannelsNameMapInCurrentTeam} from 'selectors/entities/channels';
+import {getCurrentUserId} from 'selectors/entities/users';
+
+import {GlobalState} from 'types/store';
+import {UserProfile, userProfileWithLastViewAt} from 'types/users';
+import {getDirectChannelName} from 'utils/channel_utils';
+
+export function makeAddLastViewAtToProfiles(): (state: GlobalState, profiles: UserProfile[]) => Array<userProfileWithLastViewAt> {
+    return createSelector(
+        getCurrentUserId,
+        getMyChannelMemberships,
+        getAllDirectChannelsNameMapInCurrentTeam,
```

Comment:
Why use this selector as opposed to just getting all channels here? I'm wondering if we could use a selector that does slightly less work on the channels. Also 0/5 here though

Reply:
Ya, i used simple get channels selector and made necessary changes based on it